jspdf.js

```diff
@@ -1118,15 +1118,16 @@ var jsPDF = (function(global) {
       },
       'getTextColor': function getTextColor() {
         var colorEncoded = textColor.split(' ')
-        if (colorEncoded.length == 2 && colorEncoded[-1] == 'g') {
-          return '#000000'
-        } else {
-          var x;
-          var colorAsHex = '#'
-          for (var i = 0; i < 3; i++) {
-            x = Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16);
-            colorAsHex += (x.length == 1) ? "0"+x : x;
-          }
+        if (colorEncoded.length == 2 && colorEncoded[colorEncoded.length-1] == 'g') {
+          // convert grayscale value to rgb so that it can be converted to hex for consistency
+          var floatVal = parseFloat(colorEncoded[0])
+          colorEncoded = [floatVal, floatVal, floatVal, 'r']
+        }
+        var x;
+        var colorAsHex = '#'
+        for (var i = 0; i < 3; i++) {
+          x = Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16);
```

Comment:
Probably better to merge these two lines to 

`colorAsHex += padd2(Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16));`

Reply:
Looks like padd2 won't work because hex codes can be alphanumeric and padd2 calls parseInt:
```
    padd2 = function padd2(number) {
      return ('0' + parseInt(number)).slice(-2);
    },
```

I'll still work on making this a one-liner though, I agree it's a good idea